docs/glossary.md

```diff
@@ -0,0 +1,17 @@
+# Glossary
+
+**Playlist**: This is a representation of an HLS or DASH manifest.
+
+**Master Playlist Controller**: This acts as the main controller for the playback engine. It interacts with the SegmentLoaders, PlaylistLoaders, PlaybackWatcher, etc.
+
+**Playlist Loader**: This will request the source and load the master manifest. It also interacts with the ABR algorithm to pick a media playlist or wraps a media playlist if it is provided as the source. There are more details about the playlist loader [here](./arch.md).
+
+**Media Playlist**: This is a manifest that represents a single rendition of the source.
+
+**Segment Loader**: This controls the requesting and appending of segments on to the browser's [SourceBuffers](https://developer.mozilla.org/en-US/docs/Web/API/SourceBuffer).
```

Comment:
Technically Segment Loader figures out which segment to request, and requests it via Media Segment Request. It then decrypts (if necessary) and sends the segment bytes to the Source Updater, which manages the source buffers.